recipe/session/sessionwithjwt/sessionContainer.go

```diff
@@ -21,49 +21,121 @@ import (
+			return originalSessionClass.GetSessionDataWithContext(userContext)
+		},
+
+		UpdateSessionDataWithContext: func(newSessionData map[string]interface{}, userContext supertokens.UserContext) error {
+			return originalSessionClass.UpdateSessionDataWithContext(newSessionData, userContext)
+		},
+		GetUserIDWithContext: func(userContext supertokens.UserContext) string {
+			return originalSessionClass.GetUserIDWithContext(userContext)
+		},
+		GetAccessTokenPayloadWithContext: func(userContext supertokens.UserContext) map[string]interface{} {
+			return originalSessionClass.GetAccessTokenPayloadWithContext(userContext)
+		},
+		GetHandleWithContext: func(userContext supertokens.UserContext) string {
+			return originalSessionClass.GetHandleWithContext(userContext)
+		},
+		GetAccessTokenWithContext: func(userContext supertokens.UserContext) string {
+			return originalSessionClass.GetAccessTokenWithContext(userContext)
+		},
+		GetTimeCreatedWithContext: func(userContext supertokens.UserContext) (uint64, error) {
+			return originalSessionClass.GetTimeCreatedWithContext(userContext)
+		},
+		GetExpiryWithContext: func(userContext supertokens.UserContext) (uint64, error) {
+			return originalSessionClass.GetExpiryWithContext(userContext)
+		},
+		UpdateAccessTokenPayloadWithContext: func(newAccessTokenPayload map[string]interface{}, userContext supertokens.UserContext) error {
+			if newAccessTokenPayload == nil {
+				newAccessTokenPayload = map[string]interface{}{}
+			}
+			accessTokenPayload := originalSessionClass.GetAccessTokenPayloadWithContext(userContext)
+			jwtPropertyName, ok := accessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY]
+
+			if !ok {
+				return originalSessionClass.UpdateAccessTokenPayloadWithContext(newAccessTokenPayload, userContext)
+			}
+
+			existingJWT := accessTokenPayload[jwtPropertyName.(string)].(string)
+
+			currentTimeInSeconds := uint64(time.Now().UnixNano() / 1000000000) // time in seconds
+
+			claims := jwt.MapClaims{}
+			decodedPayload := map[string]interface{}{}
+			_, _, err := new(jwt.Parser).ParseUnverified(existingJWT, claims)
+			if err != nil {
+				return err
+			}
+			for key, val := range claims {
+				decodedPayload[key] = val
+			}
+
+			jwtExpiry := uint64(decodedPayload["exp"].(float64)) - currentTimeInSeconds
+
+			if jwtExpiry <= 0 {
+				// it can come here if someone calls this function well after
+				// the access token and the jwt payload have expired (which can happen if an API takes a VERY long time). In this case, we still want the jwt payload to update, but the resulting JWT should
+				// not be alive for too long (since it's expired already). So we set it to
+				// 1 second lifetime.
+				jwtExpiry = 1
+			}
+
+			newAccessTokenPayload, err = addJWTToAccessTokenPayload(newAccessTokenPayload, jwtExpiry, originalSessionClass.GetUserIDWithContext(userContext), jwtPropertyName.(string), openidRecipeImplementation, &map[string]interface{}{})
```

Comment:
Why not pass `userContext` to this function too instead of `&map[string]interface{}{}`?